src/db-wrapper.ts

```diff
@@ -0,0 +1,233 @@
+import { Database, aql } from "arangojs";
+import { DocumentCollection } from "arangojs/collection";
+import { Config } from "arangojs/connection";
+
+import { BMT, BMT_IDs } from "./common-types";
+import { RangerConfig } from "./ranger-config";
+
+export interface IDbWrapper {
```

Comment:
I suggest to separated single db class into three more specific: 
1) Base BmtDb with method common to ordered and unordered dbs.
2) UnorderedBmt extends BmtDb and introduce methods specific to unordered db.
3) OrderedBmt extends BmtDb and introduce methods specific to ordered db.

Reply:
Isn't it a overcomplication? These db are identical by structure, and only the purpose differ.